apps/nextjs-website/src/lib/strapi/codecs/GuideCodec.ts

```diff
@@ -0,0 +1,36 @@
+import * as t from 'io-ts/lib';
+import { MediaCodec } from './MediaCodec';
+import { BaseProductCodec } from './ProductCodec';
+
+const VersionCodec = t.strict({
+  main: t.boolean,
+  dirname: t.string,
+  version: t.string,
+});
+
+export const GuideBaseAttributesCodec = t.strict({
+  id: t.number,
+  attributes: t.strict({
+    title: t.string,
+    slug: t.string,
+    image: t.strict({ data: MediaCodec }),
+    mobileImage: t.strict({ data: MediaCodec }),
+    listItems: t.array(
+      t.strict({
+        text: t.string,
+      })
+    ),
+  }),
+});
+
+const GuideRelationsCodec = t.strict({
+  attributes: t.strict({
+    versions: t.array(VersionCodec),
+    product: t.strict({ data: BaseProductCodec }),
+  }),
+});
+
+export const GuideCodec = t.intersection([
+  GuideBaseAttributesCodec,
+  GuideRelationsCodec,
+]);
```

Comment:
If the constant `GuideRelationsCodec` doesn't need to be exported, maybe you could avoid it (and its inappropriate name) by doing so:

```suggestion
export const GuideCodec = t.intersection([
  GuideBaseAttributesCodec,
  t.strict({
    attributes: t.strict({
      versions: t.array(VersionCodec),
      product: t.strict({ data: BaseProductCodec }),
    }),
  })
]);
```

Maybe we could do the same in `apps/nextjs-website/src/lib/strapi/codecs/SolutionCodec.ts`